### PR TITLE
Issue 9205 - layer list as UI for CORs enabled WMS servers

### PR DESCRIPTION
**Pull request by corrigancd**

### CODE REVIEW

**Comment by corrigancd** [comment]
FBEF072B-6239-4A7B-86CD-FCBE4179B3BA

File: public/vis.less [hunk 1]
```diff
@@ -106,6 +106,26 @@ div.rightCorner {
   color: rgb(206, 206, 206);
 }
 
+.leaflet-control-layers-toggle {
```
[the remark this answers is not in the document]
Comment (corrigancd) [reply public/vis.less]: As per Slack message, will rebase when other PR is reviewed

File: public/directives/wmsOverlay.js [hunk 2]
```diff
@@ -12,15 +13,128 @@ define(function (require) {
...
+
+    function getWMSLayerList(url) {
+      const wmsLayerNames = [];
+      const getCapabilitiesRequest = url + 'service=wms&request=GetCapabilities';
+
+      return $http.get(getCapabilitiesRequest)
+        .then(resp => {
+          if (resp.data) {
+            const wmsCapabilities = resp.data;
+            parseString(wmsCapabilities, function (err, result) {
```
[the remark this answers is not in the document]
Comment (corrigancd) [reply public/directives/wmsOverlay.js]: You're correct with the parseString, also added an "if layers" clause and tested it by creating a CORS enabled WMS with no layers, then added an additional message in the console.err.

File: public/visController.js [hunk 3]
```diff
@@ -144,7 +156,7 @@ define(function (require) {
     });
 
     $scope.$watch('esResponse', function (resp) {
-      if (_.has(resp, 'aggregations')) {
+      if (_.has(resp, 'aggregations') && (resp.aggregations[2].doc_count > 0)) {
```
[the remark this answers is not in the document]
Comment (corrigancd) [reply public/visController.js]: I console.logged three examples (zooming into just north of Cyprus for the response logged at point number 3 in the image below) with varying document counts. It seems that the doc-count is always at this point in the response: 

image

File: public/directives/wmsOverlay.js [hunk 4]
```diff
@@ -101,40 +101,40 @@ define(function (require) {
...
+
+                if (err) {
+                  reject(err);
+                }
+
+                //handles case(s) where there are no layer names returned from the WMS
+                if (result.WMS_Capabilities.Capability[0].Layer[0].Layer) {
+                  const wmsLayerNames = result.WMS_Capabilities.Capability[0].Layer[0].Layer.map(layer => layer.Name[0]);
+                  resolve(wmsLayerNames);
+                };
```
Comment (szydan) [inline public/directives/wmsOverlay.js]: then you need an else 
```
} else {
  resolve([]);
}
```
Also no ";" after if () {}; <--

**Pull request merged**